bpython/curtsiesfrontend/repl.py

```diff
@@ -357,7 +370,9 @@ def __init__(
         )
         self.edit_keys = edit_keys.mapping_with_config(config, key_dispatch)
         logger.debug("starting parent init")
-        super().__init__(interp, config)
+        # interp is a subclass of repl.Interpreter, so it definitely,
+        # implements the methods of Interpreter!
+        super().__init__(cast("Interpreter", interp), config)
 
```

Comment:
So why not mark `interp` to be of type `repl.Interpreter`?

Reply:
@sebastinas, I've reworked this to use `repl.Interpreter` instead.